lib/beaker-puppet/install_utils/foss_utils.rb

```diff
@@ -389,45 +389,45 @@ def install_puppet_agent_on(hosts, opts = {})
 
             # If inside the Puppet VPN, install from development builds.
             if opts[:puppet_agent_version] && opts[:puppet_agent_version] != 'latest' && dev_builds_accessible_on?(host, opts[:dev_builds_url])
-              return install_puppet_agent_from_dev_builds_on(host, opts[:puppet_agent_version])
-            end
-
-            if opts[:puppet_agent_version] == 'latest'
-              opts[:puppet_collection] += '-nightly' unless opts[:puppet_collection].end_with? '-nightly'
-
-              # Since we have modified the collection, we don't want to pass `latest`
-              # in to `install_package` as the version. That'll fail. Instead, if
-              # we pass `nil`, `install_package` will just install the latest available
-              # package version from the enabled repo.
-              opts.delete(:puppet_agent_version)
-              opts.delete(:version)
-            end
-
-            case host['platform']
-            when /el-|redhat|fedora|sles|centos|cisco_/
-              package_name = 'puppet-agent'
-              package_name << "-#{opts[:puppet_agent_version]}" if opts[:puppet_agent_version]
-            when /debian|ubuntu|cumulus|huaweios/
-              package_name = 'puppet-agent'
-              package_name << "=#{opts[:puppet_agent_version]}-1#{host['platform'].codename}" if opts[:puppet_agent_version]
-            when /windows/
-              install_puppet_agent_from_msi_on(host, opts)
-            when /osx/
-              install_puppet_agent_from_dmg_on(host, opts)
+              install_puppet_agent_from_dev_builds_on(host, opts[:puppet_agent_version])
```

Comment:
It looks like you are doing an if/else block instead of the return so that we continue looping instead of exiting the method.  Can we use break instead to reduce the complexity of this block?  Also, git or something did not analyze this diff very well.

Reply:
Mentioned this in slack, we can't use `break` because we want to continue the loop for the next host. I could use `next` instead, but meh..